assignments/array-methods.js

```diff
@@ -2,82 +2,159 @@
+{ "id": 43, "first_name": "Hank", "last_name": "Zebedee", "email": "hzebedee16@ezinearticles.com", "shirt_size": "L", "company_name": "Gigashots", "donation": 241 },
+{ "id": 44, "first_name": "Jodie", "last_name": "Stawell", "email": "jstawell17@yale.edu", "shirt_size": "S", "company_name": "Jaxspan", "donation": 262 },
+{ "id": 45, "first_name": "Falito", "last_name": "Karsh", "email": "fkarsh18@pcworld.com", "shirt_size": "S", "company_name": "Mycat", "donation": 239 },
+{ "id": 46, "first_name": "Reginauld", "last_name": "Purselowe", "email": "rpurselowe19@thetimes.co.uk", "shirt_size": "L", "company_name": "Jabbersphere", "donation": 11 },
+{ "id": 47, "first_name": "Vida", "last_name": "Tydd", "email": "vtydd1a@dropbox.com", "shirt_size": "S", "company_name": "Quaxo", "donation": 55 },
+{ "id": 48, "first_name": "Anderea", "last_name": "MacGiolla Pheadair", "email": "amacgiollapheadair1b@xing.com", "shirt_size": "2XL", "company_name": "Kwimbee", "donation": 214 },
+{ "id": 49, "first_name": "Bel", "last_name": "Alway", "email": "balway1c@ow.ly", "shirt_size": "S", "company_name": "Voolia", "donation": 107 },
+{ "id": 50, "first_name": "Shell", "last_name": "Baine", "email": "sbaine1d@intel.com", "shirt_size": "M", "company_name": "Gabtype", "donation": 171 }];
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(function (index) {
+    let name = index.first_name.concat(' ', index.last_name);
+    fullName.push(name);
+})
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+
+allCaps = runners.map(function (index) {
+    return index.first_name.toUpperCase();
+})
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+largeShirts = runners.filter(function (index) {
+    return index.shirt_size === 'L';
+});
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+
+ticketPriceTotal = runners.map(function (index) {
+    return index.donation;
+});
+
+ticketPriceTotal = ticketPriceTotal.reduce(function (acc, currVal) {
+    return acc + currVal;
+})
+
 console.log(ticketPriceTotal);
```

Comment:
This works perfectly fine, using a map and a reduce, but here is another way of doing it with just the reduce:

let ticketPriceTotal = runners.reduce((donationTotal, donationAmount) => {
  return donationTotal + donationAmount.donation;
}, 0);
console.log(ticketPriceTotal);